aries_cloudagent/resolver/default/tests/test_key.py

```diff
@@ -25,10 +25,14 @@ def profile():
     yield profile
 
 
-def test_supported_methods(resolver: KeyDIDResolver):
+@pytest.mark.asyncio
+async def test_supported_methods(profile, resolver: KeyDIDResolver):
     """Test the supported_methods."""
     assert resolver.supported_methods == ["key"]
-    assert resolver.supports("key")
+    assert await resolver.supports(
+        profile,
+        "did:key:UG9saXRlbmVzcyBpcyB0aGUgZmlyc3QgdGhpbmcgcGVvcGxlIGxvc2Ugb25jZSB0aGV5IGdldCB0aGUgcG93ZXIu",
```

Comment:
If the key resolver tightened up supported DID testing with regex matching the whole key, this test would fail. It might be in our best interest to include such an improvement to the key resolver.

Reply:
On second thought, the supported DID should not be extremely restrictive. It should only be used to check if it should try to use the DID and if the DID does not validate as a proper DID, that should raise some sort of `InvalidDIDError` instead of a `DIDMethodNotSupported` error. The method was supported, the DID was just incorrect.